docs/reference/modules/discovery/bootstrapping.asciidoc

```diff
@@ -0,0 +1,110 @@
+master-eligible nodes that are started to join an existing cluster. Note that
+master-eligible nodes should use storage that persists across restarts. If they
+do not, and `cluster.initial_master_nodes` is set, and a full cluster restart
+occurs, then another brand-new cluster will form and this may result in data
+loss.
+
+For a cluster with 3 master-eligible nodes (with <<node.name,node names>>
+`master-a`, `master-b` and `master-c`) the configuration will look as follows:
+
+[source,yaml]
+--------------------------------------------------
+cluster.initial_master_nodes:
+  - master-a
+  - master-b
+  - master-c
+--------------------------------------------------
+
+Alternatively the IP addresses or hostnames (<<node.name,if node name defaults
+to the host name>>) can be used. If there is more than one Elasticsearch node
+with the same IP address or hostname then the transport ports must also be
+given to specify exactly which node is meant:
+
+[source,yaml]
+--------------------------------------------------
+cluster.initial_master_nodes:
+  - 10.0.10.101
+  - 10.0.10.102:9300
+  - 10.0.10.102:9301
+  - master-node-hostname
+--------------------------------------------------
+
+Like all node settings, it is also possible to specify the initial set of
+master nodes on the command-line that is used to start Elasticsearch:
+
+[source,bash]
+--------------------------------------------------
+$ bin/elasticsearch -Ecluster.initial_master_nodes=master-a,master-b,master-c
+--------------------------------------------------
+
+It is technically sufficient to set this on a single master-eligible node in
+the cluster, and only to mention that single node in the setting, but this
+provides no fault tolerance before the cluster has fully formed. It
+is therefore better to bootstrap using at least three master-eligible nodes.
+In any case, when specifying the list of initial master nodes, **it is vitally
+important** to configure each node with exactly the same list of nodes, to
+prevent two independent clusters from forming. Typically you will set this on
+the nodes that are mentioned in the list of initial master nodes.
+
+NOTE: In alpha releases, all listed master-eligible nodes are required to be
+  discovered before bootstrapping can take place. This requirement will be
+  relaxed in production-ready releases.
+
+WARNING: You must put exactly the same set of initial master nodes in each
+  configuration file (or leave the configuration empty) in order to be sure
+  that only a single cluster forms during bootstrapping and therefore to
+  avoid the risk of data loss.
+
+[float]
+==== Choosing a cluster name
+
+The `cluster.name` allows you to create multiple clusters which are separated
+from each other. Nodes verify that they agree on their cluster name when they
+first connect to each other, and if two nodes have different cluster names then
+they will not communicate meaningfully and will not belong to the same cluster.
```

Comment:
Not sure what "communicate meaningfully" means. Can this be omitted?  If not, what's it implying?